internal/zinc-persist/src/main/scala/sbt/internal/inc/consistent/ConsistentFileAnalysisStore.scala

```diff
@@ -0,0 +1,64 @@
+// Based on zinc's FileAnalysisStore:
+package sbt.internal.inc.consistent
+
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright Lightbend, Inc. and Mark Harrah
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+import java.io.{ File, FileInputStream, FileOutputStream }
+import java.util.Optional
+
+import sbt.io.{ IO, Using }
+import xsbti.compile.{ AnalysisContents, AnalysisStore => XAnalysisStore }
+import scala.util.control.Exception.allCatch
+
+import xsbti.compile.analysis.ReadWriteMappers
+
+object ConsistentFileAnalysisStore {
+  def text(file: File, mappers: ReadWriteMappers, sort: Boolean = true): XAnalysisStore =
+    new AStore(file, new ConsistentAnalysisFormat(mappers, sort), SerializerFactory.text)
+
+  def binary(file: File, mappers: ReadWriteMappers, sort: Boolean = true): XAnalysisStore =
+    new AStore(file, new ConsistentAnalysisFormat(mappers, sort), SerializerFactory.binary)
+
+  private final class AStore[S <: Serializer, D <: Deserializer](
+      file: File,
+      format: ConsistentAnalysisFormat,
+      sf: SerializerFactory[S, D]
+  ) extends XAnalysisStore {
+
+    def set(analysisContents: AnalysisContents): Unit = {
+      val analysis = analysisContents.getAnalysis
+      val setup = analysisContents.getMiniSetup
+      val tmpAnalysisFile = File.createTempFile(file.getName, ".tmp")
+      if (!file.getParentFile.exists()) file.getParentFile.mkdirs()
+      val fout = new FileOutputStream(tmpAnalysisFile)
+      try {
+        val gout = new ParallelGzipOutputStream(fout)
```

Comment:
I'd be curious to benchmark how much of the write speedup is coming from parallelism. Would that basically require more CPU power, or assume there's excess CPU cores that aren't pinned by other tasks? In Bazel, compilation is out-sourced a worker process (and sometimes remote worker machines), but this could behave differently for sbt. Not that sbt can always occupy all cores, so we might still see speedup but it might be good to benchmark this characteristic.